examples/getitem/src/lib.rs

```diff
@@ -4,27 +4,50 @@ use pyo3::prelude::*;
 use pyo3::types::PySlice;
 
 #[pyclass]
-struct ExampleContainer {}
+struct ExampleContainer {
+    // represent the maximum length our container is pretending to be
+    max_length: i32,
+}
 
 #[pymethods]
 impl ExampleContainer {
     #[new]
     fn new() -> Self {
-        ExampleContainer {}
+        ExampleContainer { max_length: 100 }
     }
 
     fn __getitem__(&self, key: &PyAny) -> PyResult<i32> {
         if let Ok(position) = key.extract::<i32>() {
             return Ok(position);
         } else if let Ok(slice) = key.downcast::<PySlice>() {
+            // METHOD 1 - the use PySliceIndices to help with bounds checking and for cases when only start or end are provided
+            // in this case the start/stop/step all filled in to give valid values based on the max_length given
+            let indx = slice.indices(self.max_length).unwrap();
+            let delta = indx.stop - indx.start;
+            println!("Delta: {}", delta);
+
+            // METHOD 2 - Do the getattr manully really only needed if you have some special cases for stop/_step not being present
+            // convert to indices and this will help you deal with stop being the max length
             let start: i32 = slice.getattr("start")?.extract()?;
+            // This particular example assumes stop is present, but note that if not present, this will cause us to return due to the
+            // extract failing. Not needing custom code to deal with this is a good reason to use the Indices method.
             let stop: i32 = slice.getattr("stop")?.extract()?;
+            // example of grabbing step since it is not always present
+            let _step: i32 = match slice.getattr("step")?.extract() {
```

Comment:
I think this is forwarding the wrong error, i.e. `getattr` will fail if the attribute is not present, not `extract`.

Reply:
I believe the attribute is always present but the value is None